### PR TITLE
homework6

### PR DESCRIPTION
**Pull request by manproai**
Restructured BE also

### CODE REVIEW

File: client/src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+  styleUrls: ['./app.component.scss'],
 })
 export class AppComponent {
   title = 'client';
+  constructor(private authService: AuthService, private router: Router){}
+
+  get isLogged() {
+    return this.authService.isLogged();
+  }
+  logout(){
```
Comment (AudriusBaltusis) [inline client/src/app/app.component.ts:17]: please add return type

File: client/src/app/auth/auth-form/auth-form.component.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  loginForm: FormGroup = new FormGroup({
+    email: new FormControl('', [Validators.required, Validators.email]),
+    password: new FormControl('', [
+      Validators.required,
+      Validators.minLength(8),
+    ]),
+  });
+
+
+  onSubmit() {
```
Comment (AudriusBaltusis) [inline client/src/app/auth/auth-form/auth-form.component.ts:22]: please add return type everywhere

File: client/src/app/auth/services/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    return false;
+  }
+
+  logout() {
+    localStorage.removeItem('expiresIn');
+    localStorage.removeItem('idToken');
+  }
+
+  private setSesstion(res: any) {
```
Comment (AudriusBaltusis) [inline client/src/app/auth/services/auth.service.ts:34]: Please do no use `any`

File: client/src/app/todo/todo/todo.component.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-todo',
+  templateUrl: './todo.component.html',
+  styleUrls: ['./todo.component.scss']
+})
+export class TodoComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (AudriusBaltusis) [inline client/src/app/todo/todo/todo.component.ts:12]: Please remove empty `ngOnInit`

File: client/src/app/app.component.ts [hunk 5]
```diff
@@ -1,10 +1,23 @@
...
   selector: 'app-root',
   templateUrl: './app.component.html',
-  styleUrls: ['./app.component.scss']
+  styleUrls: ['./app.component.scss'],
 })
 export class AppComponent {
   title = 'client';
+  constructor(private authService: AuthService, private router: Router){}
+
+  get isLogged() {
```
Comment (AudriusBaltusis) [inline client/src/app/app.component.ts:14]: Please add type

**Review by AudriusBaltusis — CHANGES_REQUESTED** [review]

File: client/src/app/auth/services/auth.service.ts (hunk 3 above)
Comment (manproai) [reply client/src/app/auth/services/auth.service.ts:34]: Should I provide all the possibles types it receives if I do not use any in this case. I tried to provide all the types but it did not compile, when I log type of res, it showed as an object but if I put object as "res: object", it is not compiling.

**Pull request closed without merging**